R-package/R/xgb.DMatrix.R

```diff
@@ -528,6 +528,135 @@ setinfo.xgb.DMatrix <- function(object, name, info, ...) {
   stop("setinfo: unknown info name ", name)
 }
 
+#' @title Get Quantile Cuts from DMatrix
```

Comment:
Is it necessary to return different formats? I think it might be better off to pick one that's most efficient and let the users handle the rest of the conversion.

Reply:
I think its good to have a user-friendly format and a more efficient format depending on the use-case.

Reply:
I'm not sure `csc` is a good format for this return, 0 is a valid value in cuts. XGBoost can split with 0 as the threshold.

Reply:
These R CSC matrices can have explicit zeros when one creates them by manually assigning to the slots instead of using the checked constructors.

Reply:
The problem is extra zeros are padded into the end.

Reply:
It's not exactly a CSC matrix, or a matrix at all. I'm not comfortable with coercing it into one without good use cases.

Reply:
Yes, that's why I set the "list" format to be the default.

Reply:
My idea for this CSC format was to be able to use the `print` method from `Matrix`, which displays the information in a way that's visually easier to parse.

Reply:
Is it necessary to be done in this function? I prefer using helper functions for wrangling data. XGBoost itself should try return its data with minimal ambiguity.

Reply:
Removed this CSC format.